File: Divide-Conquer/Z/1074.py
def divide(size, y, x):
    print(size, y, x)
    if y == r and x == c:
        print(ans)
        return
    if size == 1:
        ans += 1
        return
    if not (y<=r and r<y+size and x<=c and c<x+size):
        ans = size*size
        return

    half = size//2
    divide(half, y, x)
    divide(half, y, x+half)
    divide(half, y+half, x)
    divide(half, y+half, x+half)



n, r, c = map(int, input().split())
global ans
ans = 0
divide(2**n, 0, 0)

File: Divide-Conquer/Z/1074.cpp
#include <iostream>
#include <cmath>

using namespace std;

int N, R, C;
int ans = 0;

void divide(int size, int y, int x) {
    cout << size << " " << y << " " << x << endl;
    
    if (y == R && x == C){
        cout << ans << endl;
        return
    }

}

int main() {
    cin >> N >> R >> C;
    divide(pow(2, N), 0, 0);
}
